core/src/main/java/fi/iki/elonen/NanoHTTPD.java

```diff
@@ -1196,6 +1196,22 @@ private void decodeHeader(BufferedReader in, Map<String, String> pre, Map<String
                 System.out.println(this + " - decodeHeader: " + inLine);
                 
                 StringTokenizer st = new StringTokenizer(inLine);
+                
+                // check if we are dealing with a request that went through the proxy protocol
+                String proxy = inLine.substring(0, 5).toLowerCase();
+                if (proxy.equals("proxy")) {
+                	System.out.println(this + " - decodeHeader: detected proxy protocol.");
+                	
+                	// throw away the PROXY part
+                	st.nextToken();
+                	headers.put("x-proxy-protocol", st.nextToken());
+                	headers.put("x-proxy-client-ip", st.nextToken());
+                	headers.put("x-proxy-proxy-ip", st.nextToken());
+                	headers.put("x-proxy-client-port", st.nextToken());
+                	headers.put("x-proxy-proxy-port", st.nextToken());
```

Comment:
Wondering if we should make the other fabricated/injected headers we have there follow the same convention with `x-` prefix and modify the servers (airplay, fairplay and web) to take that into consideration when looking for those headers?